File: csnet/commons/src/org/vpc/neormf/commons/types/converters/IntegerToBoolean.cs
using System;
using org.vpc.neormf.commons.sql;
/**
 * class presentation
 *
 * @author taha BEN SALAH (tbensalah)
 * @version 1.0
 * @copyrights (c) 2004, Vpc Open Source Foundary
 * @project New Entreprise Object Relational Mapping Framework (neormf)
 * @creation on Date: 2 avr. 2004 Time: 14:41:43
 * @modification on ---- by -----
 * @modification on ---- by -----
 * @modification on ---- by -----
 */
namespace org.vpc.neormf.commons.types.converters
{
	public class IntegerToBoolean : DataTypeConverter 
	{
		public static DataTypeConverter INTEGER_TO_BOOLEAN = new IntegerToBoolean(false);

		private DataType sql = new IntType(true);
		private DataType business = BooleanType.BOOL_NULLABLE;

		private bool nullIsFalse;

		public IntegerToBoolean() :this(false)
		{
			;
		}
    
		public IntegerToBoolean(bool nullIsFalse) 
		{
			this.nullIsFalse=nullIsFalse;
		}

		public override DataType GetSQLDataType() 
		{
			return sql;
		}

		public override DataType GetBusinessDataType() 
		{
			return business;
		}

		public override Object BusinessToSQL(Object businessObject) 
		{
			if(nullIsFalse)
			{
				if(businessObject == null)
				{
					return 0;
				}
				return ((Boolean) businessObject) ? 1 : 0;
			}
			else
			{
				if(businessObject == null)
				{
					return null;
				}
				return  ((Boolean) businessObject) ? 1 : 0;
			}
		}

		public override Object SqlToBusiness(Object sqlObject) 
		{
			if(sqlObject == null || sqlObject is DBNull)
			{
				if(nullIsFalse)
				{
					return false;
				}
				return null;
			}
			if(sqlObject.Equals(1))
			{
				return true;
			}
			if(sqlObject.Equals(0))
			{
				return false;
			}
			throw new NoSuchElementException("Unexpected sql value '"+sqlObject+"' ; possible values are {0,1}");
			//        return sqlObject == null ? null : trueChar.Equals(sqlObject) ? Boolean.TRUE : falseChar.Equals(sqlObject) ? Boolean.FALSE
			//                : null; // throw exception ???
		}
	}
}

File: java/commons/src/main/java/org/vpc/neormf/commons/types/converters/IntegerToBoolean.java
/*
 * CopyrightPlugin (c) 2004 Your Corporation. All Rights Reserved.
 */
package org.vpc.neormf.commons.types.converters;

import org.vpc.neormf.commons.types.BooleanType;
import org.vpc.neormf.commons.types.DataType;
import org.vpc.neormf.commons.types.IntType;

import java.util.NoSuchElementException;

/**
 * class presentation
 *
 * @author taha BEN SALAH (tbensalah)
 * @version 1.0
 * @copyrights (c) 2004, Vpc Open Source Foundary
 * @project New Entreprise Object Relational Mapping Framework (neormf)
 * @creation on Date: 2 avr. 2004 Time: 14:41:43
 * @modification on ---- by -----
 * @modification on ---- by -----
 * @modification on ---- by -----
 */
public class IntegerToBoolean extends DataTypeConverter {

    private DataType sql = new IntType(true);

    private boolean nullIsFalse;

    public IntegerToBoolean() {
        this(true);
    }
    
    public IntegerToBoolean(boolean nullIsFalse) {
        this.nullIsFalse=nullIsFalse;
    }

    public DataType getSQLDataType() {
        return sql;
    }

    public DataType getBusinessDataType(DataType baseType) {
        return baseType.isNullable()?BooleanType.BOOL_NULLABLE:BooleanType.BOOL_NON_NULLABLE;
    }

    public Object businessToSQL(Object businessObject) {
        if(nullIsFalse){
            return businessObject == null ? new Integer(0) : ((Boolean) businessObject).booleanValue() ? new Integer(1) : new Integer(0);
        }else{
            return businessObject == null ? null : ((Boolean) businessObject).booleanValue() ? new Integer(1) : new Integer(0);
        }
    }

    public Object sqlToBusiness(Object sqlObject) {
        if(sqlObject == null){
            return nullIsFalse?Boolean.FALSE: null;
        }
        if(new Integer(1).equals(sqlObject)){
            return Boolean.TRUE;
        }
        if(new Integer(0).equals(sqlObject)){
            return Boolean.FALSE;
        }
        throw new NoSuchElementException("Unexpected sql value '"+sqlObject+"' ; possible values are {0,1}");
//        return sqlObject == null ? null : trueChar.equals(sqlObject) ? Boolean.TRUE : falseChar.equals(sqlObject) ? Boolean.FALSE
//                : null; // throw exception ???
    }
}
